test/session/sessionUtils.test.js

```diff
@@ -0,0 +1,53 @@
+/* Copyright (c) 2024, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+const assert = require("assert");
+const { normaliseSessionScopeOrThrowError } = require("../../lib/build/recipe/session/utils");
+
+describe("normaliseSessionScopeOrThrowError", () => {
```

Comment:
also add tests with sub domains like:
- `sub.example.com` should return `sub.example.com`
- `http://sub.example.com` should return `sub.example.com` 
- `https://sub.example.com` should return `sub.example.com` 
- `.sub.example.com` should return `.sub.example.com`
- `a.sub.example.com` should return `a.sub.example.com`
- `http://a.sub.example.com` should return `a.sub.example.com` 
- `https://a.sub.example.com` should return `a.sub.example.com` 
- `.a.sub.example.com` should return `.a.sub.example.com`